lib/active_model/serializer.rb

```diff
@@ -17,32 +19,10 @@ class Serializer
     include Associations
     include Attributes
     include Caching
+    include Links
+    include Type
     require 'active_model/serializer/adapter'
 
-    with_options instance_writer: false, instance_reader: false do |serializer|
-      serializer.class_attribute :_type, instance_reader: true
-      serializer.class_attribute :_links # @api private : links definitions, @see Serializer#link
-      self._links ||= {}
-    end
-
-    # Serializers inherit _attribute_mappings, _reflections, and _links.
-    # Generates a unique digest for each serializer at load.
```

Comment:
yeah.. just makes it harder to chase down by inspecting the code... really just thinking about readability